routing/endpointregistry_test.go

```diff
@@ -48,6 +49,29 @@ func TestSetAndGetAnotherKey(t *testing.T) {
 	assert.Equal(t, int64(1), mToChange.InflightRequests())
 }
 
+func TestDoRemovesOldEntries(t *testing.T) {
+	route := &Route{LBEndpoints: []LBEndpoint{
+		{Host: "existing endpoint"},
+	}}
+	route.BackendType = eskip.LBBackend
+
+	r := NewEndpointRegistry(RegistryOptions{})
+
+	r.IncInflightRequest("existing endpoint")
+	r.IncInflightRequest("removed endpoint")
+	r.lastSeen["removed endpoint"] = time.Now().Add(-lastSeenTimeout)
```

Comment:
This tests internals and would break after refactoring.
Ideally we should test behavior, e.g.:
1. r.Do(<route with endpoint1 and endpoint2>)
2. e.g. check detected time of endpoint1 and endpoint2
3. wait (this could be done by moving "now" function and moving clock forward)
4. r.Do(<route with endpoint1 only>)
5. check that endpoint2 was removed e.g. by checking its detected time